src/utils/walletConnect.js

```diff
@@ -166,3 +167,53 @@ export const getWalletConnectCallRequestType = (callRequest: WalletConnectCallRe
       return REQUEST_TYPE.UNSUPPORTED;
   }
 };
+
+/**
+ * Heuristic way of parsing app name.
+ *
+ */
+export const parsePeerName = (name: ?string): string => {
+  if (!name) return '';
+
+  let result = name;
+
+  // Remove text after hyphen
+  // eslint-disable-next-line prefer-destructuring
+  result = name.split('-')[0];
+
+  // Strip all emojis
+  result = stripEmoji(result);
+
+  // Fallback if there is nothing left
+  if (result.length === 0) {
+    result = name;
+  }
+
+  result = result.trim();
+
+  // Limit length
+  if (result.length > 20) {
+    // eslint-disable-next-line i18next/no-literal-string
+    result = `${result.substring(0, 20)}…`;
+  }
+
+  return result.trim();
+};
+
+/**
+ * Heuristic way of picking the best icon.
+ *
+ * We try to pick PNG icons with highest pixel value by sorting them first by URL length (desc), then by name (desc).
+ * Otherwise just pick whatever is there. See test file for sample cases.
+ */
+export function pickPeerIcon(icons: ?string[]): ?string {
+  if (!icons?.length) return null;
+  if (icons?.length === 1) return icons[0];
+
+  const pngUrls = icons.filter((url) => url.endsWith('.png'));
+  if (!pngUrls.length) return icons[1];
```

Comment:
Maybe you meant `[0]`? Or if there are more than 1 icon in array second icon is always something different? If so - a note would help 🙏

Reply:
Added comment